Modules/rat_facts.py

```diff
@@ -0,0 +1,187 @@
+"""
+rat_facts.py - Managing Facts since 2018
+
+Copyright (c) 2018 The Fuel Rat Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+
+This module is built on top of the Pydle system.
+
+"""
+from utils.ratlib import Singleton
+from Modules.database_manager import DatabaseManager
+from Modules.context import Context
+import re
+import pyodbc
+import datetime
+
+internal_facts = ("!go", "!assign", "!add")
+# any fact name in here will not be handled by the handler
+
+
+class Fact:
+    name = None
+    language = None
+    message = None
+    author = None
+    timestamp = None
+
+    def __init__(self, name: str, lang: str, message: str, author: str,
+                 timestamp: datetime.datetime):
+        self.name = name
+        self.lang = lang
+        self.message = message
+        self.author = author
+        self.timestamp = timestamp
+
+
+class FactsManager(Singleton):
+
+    @property
+    def enabled(self):
+        return self.dbm.enabled
+
+    @enabled.setter
+    def enabled(self, value):
```

Comment:
Then don't expose a setter if you don't want people setting it 😉 
Python will raise an `AttributeError` if someone tries setting a property that does not have a defined setter method.

Reply:
It's merely a debugging aid, but I can remove it too ;)

Reply:
removed